chickadee/io.py

```diff
@@ -49,10 +49,230 @@
 )
 
-end_date = LiteralInput(
-    "end_date",
-    "End Date",
-    abstract="Defines the end of the calibration period",
-    default=date(2005, 12, 31),
-    data_type="date",
-)
+
+general_options_input = [
+    LiteralInput(
+        "units_bool",
+        "Units Boolean",
+        abstract="Whether to check the input units and convertthem to the target output units",
+        max_occurs=1,
+        default=True,
+        data_type="boolean",
+    ),
+    LiteralInput(
+        "n_pr_bool",
+        "Negative Precipitation Boolean",
+        abstract="Whether to check for and eliminate negativeprecipitation values",
+        max_occurs=1,
+        default=True,
+        data_type="boolean",
+    ),
+    LiteralInput(
+        "tasmax_units",
+        "Tasmax Units",
+        abstract="Units used for tasmax in output file",
+        max_occurs=1,
+        default="celsius",
+        data_type="string",
+    ),
+    LiteralInput(
+        "tasmin_units",
+        "Tasmin Units",
+        abstract="Units used for tasmin in output file",
+        max_occurs=1,
+        default="celsius",
+        data_type="string",
+    ),
+    LiteralInput(
+        "pr_units",
+        "Precipitation Units",
+        abstract="Units used for pr in output file",
+        max_occurs=1,
+        default="kg m-2 d-1",
+        data_type="string",
+    ),
+    LiteralInput(
+        "max_gb",
+        "Max GB",
+        abstract="Anapproximate measure of how much RAM to use in the chunk I/O loop. It’s best to set this to about 1/3 to 1/4 of what you want the high-water mark to be",
+        max_occurs=1,
+        default=1.0,
+        data_type="float",
+    ),
+    LiteralInput(
+        "start_date",
+        "Start Date",
+        abstract="Defines the stat of the calibration period",
+        max_occurs=1,
+        default=date(1971, 1, 1),
+        data_type="date",
+    ),
+    LiteralInput(
+        "end_date",
+        "End Date",
+        abstract="Defines the end of the calibration period",
+        max_occurs=1,
+        default=date(2005, 12, 31),
+        data_type="date",
+    ),
+]
+
+
+ca_options_input = [
+    LiteralInput(
+        "num_analogues",
+        "Number of Analogues",
+        abstract="The number of temporal analogues that the CA algorithm will search for andmatch.  The higher this number, the longer the execution time of the reorderingstep.",
+        max_occurs=1,
+        default=30,
+        data_type="positiveInteger",
+    ),
+    LiteralInput(
+        "delta_days",
+        "Delta Days",
+        abstract="An integer describing the size of the window on either side of a day - Not recommeded to change",
+        max_occurs=1,
+        default=45,
+        data_type="positiveInteger",
+    ),
+    LiteralInput(
+        "trimmed_mean",
+        "Trimmed Mean",
+        abstract="The fraction (0 to 0.5) of observations to be trimmed from each end of the set of values before the mean is computed - Not recommended to change",
```

Comment:
This is fairly clear, _maybe_ we could use slightly more "statistical" language, like:

"The fraction (0 to 0.5) of observations to be trimmed from each extreme of the distribution before the mean is computed - Not recommended to change"